crates/bevy_gizmos/src/circles.rs

```diff
@@ -266,3 +305,67 @@ where
         self.gizmos.linestrip_2d(positions, self.color);
     }
 }
+
+/// Builder for configuring the drawing options of [`Sphere`].
+pub struct SphereBuilder<'a, 'w, 's, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    gizmos: &'a mut Gizmos<'w, 's, Config, Clear>,
+
+    // Radius of the sphere
+    radius: f32,
+
+    // Rotation of the sphere around the origin in 3D space
+    rotation: Quat,
+    // Center position of the sphere in 3D space
+    position: Vec3,
+    // Color of the sphere
+    color: Color,
+
+    // Number of segments used to approximate the sphere geometry
+    segments: usize,
+}
+
+impl<Config, Clear> SphereBuilder<'_, '_, '_, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    /// Set the number of segments used to approximate the sphere geometry.
+    pub fn segments(mut self, segments: usize) -> Self {
+        self.segments = segments;
+        self
+    }
+}
+
+impl<Config, Clear> Drop for SphereBuilder<'_, '_, '_, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    fn drop(&mut self) {
+        if !self.gizmos.enabled {
+            return;
+        }
+
+        let SphereBuilder {
+            radius,
+            position: center,
+            rotation,
+            color,
+            segments,
+            ..
+        } = self;
+
+        // draws one great circle around each of the local axes
+        let rotation = rotation.normalize();
+        Vec3::AXES.into_iter().for_each(|axis| {
+            let normal = rotation * axis;
+            self.gizmos
+                .circle(*center, Dir3::new_unchecked(normal), *radius, *color)
+                .segments(*segments);
+        });
```

Comment:
Looks like this was changed from a circle per segment to a circle per axis to match the "non-primitive" sphere gizmo. I definitely prefer this since the old version was kinda excessive and probably expensive with lots of spheres. Might be worth mentioning in the PR description though?

I think it'd also be good to do this for some other primitives like capsules, but that can be another follow-up.

Reply:
I've done that for cylinders and capsules aswell now :)